samza-core/src/main/java/org/apache/samza/container/host/LinuxCgroupStatisticsGetter.java

```diff
@@ -0,0 +1,133 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.container.host;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import java.io.FileReader;
+import java.nio.file.Files;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.conf.Configuration;
+import java.net.URL;
+import java.net.URI;
+import java.net.MalformedURLException;
+import java.util.Optional;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.io.File;
+
+public class LinuxCgroupStatisticsGetter implements SystemStatisticsGetter {
+  private static final Logger LOG = LoggerFactory.getLogger(LinuxCgroupStatisticsGetter.class.getName());
+  private String cgroupMountPath;
+  private String cgroupHierarchy;
+  private String containerID;
+
+  @Override
+  public SystemMemoryStatistics getSystemMemoryStatistics() {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  @Override
+  public ProcessCPUStatistics getProcessCPUStatistics() {
+    throw new UnsupportedOperationException("Not implemented");
+  }
+
+  LinuxCgroupStatisticsGetter() {
+    this.containerID = Optional.ofNullable(System.getenv("CONTAINER_ID")).orElse("NOT_DETECTED");
+    String hadoopConfDir =  Optional.ofNullable(System.getenv("HADOOP_CONF_DIR")).orElse("NOT_DETECTED");
+    Configuration yarnSite;
+    if (!hadoopConfDir.equals("NOT_DETECTED")) {
+      yarnSite = getHadoopConf(hadoopConfDir);
+      this.cgroupHierarchy = yarnSite.get("yarn.nodemanager.linux-container-executor.cgroups.hierarchy", "NOT_DETECTED");
+      this.cgroupMountPath = yarnSite.get("yarn.nodemanager.linux-container-executor.cgroups.mount-path", "NOT_DETECTED");
+    }
+    LOG.debug("CONTAINER ID: " + this.containerID);
+    LOG.debug("HADOOP_CONF_DIR: " + hadoopConfDir);
+    LOG.debug("CGROUP_MOUNT_PATH: " + this.cgroupMountPath);
+    LOG.debug("CGROUP_MOUNT_HIERARCHY: " + this.cgroupHierarchy);
+  }
+
+  @Override
+  public LinuxCgroupStatistics getProcessCgroupStatistics() {
+    try {
+      double ratio = getCPUStat();
+      return new LinuxCgroupStatistics(ratio);
+    } catch (Exception e) {
+      LOG.debug("Error reading cgroups information: ", e);
+      return null;
+    }
+  }
+
+  private double getCPUStat() {
+    if (this.containerID.equals("NOT_DETECTED")) {
```

Comment:
Where is NOT_DETECTED defined? Should this be a new const?

Reply:
Updated in the next push